lib/rouge/lexers/cmhg.rb

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class CMHG < RegexLexer
+      title "CMHG"
+      desc "RISC OS C module header generator source file"
+      tag 'cmhg'
+      filenames '*.cmhg'
+
+      def self.preproc_keyword
+        @preproc_keyword ||= %w(
+          define elif else endif error if ifdef ifndef include line pragma undef warning
+        )
+      end
+
+      state :root do
+        rule %r/;[^\n]*/, Comment
+        rule %r/^[ \t]*#[ \t]*(?:(?:#{CMHG.preproc_keyword.join('|')})[ \t].*)?/, Comment::Preproc
```

Comment:
So it's possible for a preprocessor command just to be a `#`?

Reply:
If the rule does need to begin with `[ \t]*`, I'd suggest using a do-block and tokenising this capture group as `Text` rather than `Comment::Preproc`.

Reply:
(Note: this rule was a cut-and-paste from my ArmAsm lexer, and any issues probably apply equally there.)

The file format is (optionally) run through the C preprocessor before the CMHG tool does its own main lexing process. The C preprocessor does accept a lone `#` character on a line, yes, and converts any such line into a blank line. In fact, `#` is not a valid character anywhere in the underlying syntax here (after preprocessing).

The underlying syntax treats `\n` the same as any other type of whitespace, but the C preprocessor is line-oriented, which is why I search for `#` being the first non-whitespace character of a line. I'm a little surprised that you suggest it is tokenised as `Text`, when `Comment::Preproc` is what is used in the C lexer. Surely these should at least be consistent?

Reply:
Ah, I'm sorry, my comment was a little unclear. What I meant to suggest was that the whitespace should be tokenised separately from the preprocessor command. This can be done like this:

```ruby
rule %r/^([ \t]*)(#[ \t]*(?:(?:#{CMHG.preproc_keyword.join('|')})(?:[ \t].*)?)?)(\n)/ do
  groups Text, Comment::Preproc, Text
end
```

And yes, it was silly for me not to notice this issue in the ArmAsm lexer. _Especially_ not to notice it before I merged it into master :P I'll fix that with #1289!

Reply:
OK, gotcha. I've just pushed an equivalent fix.